src/m365/aad/commands/o365group/o365group-remove.ts

```diff
@@ -47,7 +48,25 @@ class AadO365GroupRemoveCommand extends GraphCommand {
 
       request
         .delete(requestOptions)
-        .then(_ => cb(), (rawRes: any): void => this.handleRejectedODataJsonPromise(rawRes, logger, cb));
+        .then((): void => {
+
+          if (args.options.skipRecycleBin) {
+            const requestOptions2: any = {
+              url: `${this.resource}/v1.0/directory/deletedItems/${args.options.id}`,
+              headers: {
+                'accept': 'application/json;odata.metadata=none'
+              },
+            };
+
+            request
+            .delete(requestOptions2)
+            .then((): void => {
```

Comment:
Let's avoid nesting promises and rather chain them. We should change this piece to `return request.delete()` or `return Promise.resolve()` (when no skip recycle bin was requested). It will not only help us simplify the code but also centralize handling errors. Right now, we're not handling failed requests for removing the group from the recycle bin which could lead to unhandled exceptions if the call fails.

Reply:
I changed this part. 👍